lib/propolis/src/hw/virtio/softnpu.rs

```diff
@@ -569,12 +575,16 @@ impl PacketHandler {
     }
 
     /// Send a packet out an external port using dlpi.
-    fn send_packet_to_ext_port<'a>(
-        pkt: &mut packet_out<'a>,
+    fn send_packet_to_ext_port(
+        pkt: &mut packet_out<'_>,
         data_handles: &Vec<dlpi::DlpiHandle>,
         port: u16,
         log: &Logger,
     ) {
+        if usize::from(port) >= data_handles.len() {
+            error!(log, "port out of range {} > {}", port, data_handles.len());
```

Comment:
```suggestion
            error!(log, "port out of range {} >= {}", port, data_handles.len());
```

nit: make the log output match the actual condition